### PR TITLE
Fix NA's when age<1

### PR DESCRIPTION
**Pull request by joewalker127**
Translates age values initially from scientific notation (can occur when age < 1 year in raw excel import) to standard notation to prevent NA's from arising when calling "age = as.numeric(age)"

### CODE REVIEW

**Comment by thibautjombart** [comment]
Thanks for this. Will have a look code-wise but @jarvisc1 will handle the PR in terms of functionality. The source of the error seemed to be a `05./12` in the google shit that was missing an `=` sign and thus was text, so the entire variable is promoted to `character` when imported into R and then cleaned as such by `clean_data`.

File: linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd [hunk 1]
```diff
@@ -373,8 +373,14 @@ This section defines age classes - see code for details.
 ```{r clean_age}
```
Comment (thibautjombart) [inline linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd]: Explain briefly what the code does (what problem / cases it covers) before the code chunk in plain English.

File: linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd [hunk 2]
```diff
@@ -373,8 +373,14 @@ This section defines age classes - see code for details.
...
 
 x <- x %>%
-  mutate(
-      age = as.numeric(gsub("_", ".", age)),
+  mutate(age = gsub("_", ".", age))
```
Comment (thibautjombart) [inline linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd]: This is dangerous, as it will be converting age to `character` if it is a number, e.g.:
```r
> gsub("_", ".", 1.23)
[1] "1.23"
```
Best condition the whole thing on `age` being a `character` using `if (is.character(x$age)) {...}`

File: linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd [hunk 3]
```diff
@@ -373,8 +373,14 @@ This section defines age classes - see code for details.
...
+
+# line below converts values that were initially in scientific notation (indicative of age < 1 year) to # prevent NA's from arising when calling "age = as.numeric(age)"
```
Comment (thibautjombart) [inline linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd]: use 2 ## for comments raher than #, as a single one is indented differently on different editors

File: linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd [hunk 4]
```diff
@@ -373,8 +373,14 @@ This section defines age classes - see code for details.
...
+  x$age[grep("E.",x$age)]=map_chr(.x=x$age[grep("E.",x$age)],.f=function(.x){return(as.character(as.numeric(substr(.x,1,regexpr("E.",.x)-1))*10^-as.numeric(substr(.x,regexpr("E.",.x)+2,nchar(.x)))))})
```
Comment (thibautjombart) [inline linelist_investigations/report_sources/aaa_clean_linelist_2019-10-18.Rmd]: This is hard to read. Best break this down into intermediate object, including the subsetting using grep and the function. Might be best to refer explicitely to the package for `map_chr` i.e. `purrr::map_chr` since `purrr` is not used much elsewhere. 

Style things - check the `lintr` package but in a nutshell: 

* use `<-` for assignment
* flank operators with spaces
* add space after `,`

**Comment by jarvisc1** [comment]
Looking further into the data it looks like this was cased by some strange unicode character in the original data which convert the value to character when R reads in the data. We looking at removing the unicode character and then the scientific value is read in correctly within R. In that case there won't be any strange values with 0.23.E3 so this step won't be needed. 

For the moment we'll hold off on this. Thanks for putting the PR together.

**Comment by jarvisc1** [comment]
Closed as data issue is sorted.
